modules/apps/object/object-service/src/main/java/com/liferay/object/internal/item/selector/SystemObjectEntryItemSelectorView.java

```diff
@@ -364,14 +367,23 @@ public SearchContainer<BaseModel<?>> getSearchContainer()
 
 				List<BaseModel<?>> baseModels =
 					objectRelatedModelsProvider.getUnrelatedModels(
+						_themeDisplay.getCompanyId(),
+						_themeDisplay.getScopeGroupId(), _objectDefinition,
+						ParamUtil.getLong(_portletRequest, "objectEntryId"),
+						ParamUtil.getLong(
+							_portletRequest, "objectRelationshipId"),
+						searchContainer.getStart(), searchContainer.getEnd());
+
+				int baseModelsCount =
+					objectRelatedModelsProvider.getUnrelatedModelsCount(
 						_themeDisplay.getCompanyId(),
 						_themeDisplay.getScopeGroupId(), _objectDefinition,
 						ParamUtil.getLong(_portletRequest, "objectEntryId"),
 						ParamUtil.getLong(
 							_portletRequest, "objectRelationshipId"));
 
 				searchContainer.setResultsAndTotal(
-					() -> baseModels, baseModels.size());
+					() -> baseModels, baseModelsCount);
```

Comment:
If the formatsource allow, I suggest to inline.
```suggestion
searchContainer.setResultsAndTotal(
					() -> baseModels,
					objectRelatedModelsProvider.getUnrelatedModelsCount(
						_themeDisplay.getCompanyId(),
						_themeDisplay.getScopeGroupId(), _objectDefinition,
						ParamUtil.getLong(_portletRequest, "objectEntryId"),
						ParamUtil.getLong(
							_portletRequest, "objectRelationshipId")););
```